tiatoolbox/models/architecture/vanilla.py

```diff
@@ -79,6 +80,48 @@ def _get_architecture(
     return model.features
 
 
+def _get_timm_architecture(
+    arch_name: str,
+    weights: str or WeightsEnum = "DEFAULT",
```

Comment:
weights are not required to load a model. These are provided in the `https://github.com/TissueImageAnalytics/tiatoolbox/blob/015652cc5c2357070592c8d46f5f2ff9a905a5c6/tiatoolbox/models/engine/patch_predictor.py#L231` or through `https://github.com/TissueImageAnalytics/tiatoolbox/blob/dd712ad173710bd27824e1141e34cb23f84792d1/tiatoolbox/data/pretrained_model.yaml` by default.

Reply:
Removing `weights` should also fix the tests.

Reply:
Are you suggesting to completely remove the `weights` argument from `_get_timm_architecture()`?

I am hardcoding the HuggingFace path for `timm` in `_get_timm_architecture()` as an initial version. But ideally users would be able to specify the path from where to load the `timm` model weights.

Reply:
Adding `weights` argument would make it inconsistent with the rest of the architectures and the Engines API. We deliberately kept `weights` separate from architecture definition as it allows to import different weights in different pipelines in a flexible way. Hardcoding will not allow much flexibility. Once the architecture is loaded, loading weights shouldn't be a problem through Engines.
We are carefully reviewing this in #578 and #783.

Reply:
Sorry, I confused it with the engines. I see you have made it similar to `_get_architecture()` https://github.com/TissueImageAnalytics/tiatoolbox/blob/7aa3ca3e3fb563ef89b7fae84e35b9eff803d548/tiatoolbox/models/architecture/vanilla.py#L20